### PR TITLE
Remove suite_owner option and make --colour cli help more informative

### PR DESCRIPTION
**Pull request by datamel**
This is a very small change with no associated Issue.

Removed suite owner from the cli options and listed options available to users for colour in the help.

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (removing code, tests unaffected).
- [ ] Appropriate change log entry included.
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

File: cylc/flow/option_parsers.py [hunk 1]
```diff
@@ -149,15 +149,12 @@ def add_std_options(self):
...
-            self.add_std_option(
-                "--suite-owner",
-                help="Specify suite owner",
-                metavar="OWNER", action="store", default=None,
-                dest="suite_owner")
-
+                help=(
+                    "Determine when to use color in terminal output. "
+                    "Options available are 'never', 'auto' and 'always'. "
+                    "'never' will also remove bold typeface."
```
Comment (oliver-sanders) [inline cylc/flow/option_parsers.py:155]: Bold text isn't specific to `--color=never` (awkwardly it doesn't necessarily render as bold the terminal either):

```suggestion
                    "Determine when to use color/bold text in terminal output. "
                    "Options available are 'never', 'auto' and 'always'."
```

File: cylc/flow/suite_files.py [hunk 2]
```diff
@@ -429,15 +429,13 @@ def get_workflow_source_dir(
...
+def get_suite_srv_dir(reg):
     """Return service directory of a suite."""
-    if not suite_owner:
-        suite_owner = get_user()
     run_d = os.getenv("CYLC_SUITE_RUN_DIR")
     if (
         not run_d
         or os.getenv("CYLC_SUITE_NAME") != reg
-        or os.getenv("CYLC_SUITE_OWNER") != suite_owner
+        or os.getenv("CYLC_SUITE_OWNER") != get_user()
```
Comment (oliver-sanders) [inline cylc/flow/suite_files.py:438]: If `suite_owner` was always `None` before I think we can remove it?

```suggestion
```

Comment (datamel) [reply cylc/flow/suite_files.py]: No probs, it defaulted to get_user() if it was none, just for purposes of documenting in the contact file, but I can remove the whole field if it is not a useful one.

**Comment by hjoliver** [comment]
@datamel - this PR might be an appropriate place to address #4171 ?  See what you think.

**Review by wxtim — APPROVED** [review]
Happy with change. Please can you address the style failure, then merge?

**Pull request merged**